### PR TITLE
Active Record Basics section of Rails course: Add Migrations lesson

### LINKED ISSUE #27477: Rails course, Active Record Basics lesson: Split into multiple lessons
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

### The old version and what happened

If you go [here](https://web.archive.org/web/20150207114235/http://www.theodinproject.com/ruby-on-rails/active-record-basics), you can see a much older version of how this lesson used to look. You'll see that the lesson used to have multiple reading assignments to read those docs, inserted throughout the content. Each pre-assignment reading was originally intended as some light, high-level browsing to do before diving into those docs.

At some point, the decision was made to compile all of the 'overview reading' into a single chunk, have the student read all of that introductory text first, and then have the student read all those assigned docs at once. **I'm not questioning that decision -- it was valid because its purpose (I assume) was to make the layouts of all TOP lessons consistent. That's totally fine and I respect it.**

### The main problem

That correction introduced a problem which, IMO, should now be fixed also: by the time you're reading, for example, the Migrations section, **you were supposed to have already done some docs reading!** That matters, and I think I can give some evidence that it does:

Screenshot from 2024-02-27 16-27-47

As you can see outlined in red, the lesson is stating that you should have 'just learned above' how to set up a model file. The reader of the new version thinks "ok, it's referring to that '30 seconds about working with models' section I just read". So you scroll up there to check where it says how to set up the model. Does it? To the best of my understanding, it doesn't. It says that "you create a model called User, which is really just a Ruby file which inherits from Active Record". But that's it. The rest of the section briefly demonstrates *how to add a new user, which is a new instance of the already-existing User model, not a new model*. IMO, that section doesn't include the steps for actually setting up a model file.

So, could "just learned above" have originally been a reference to the **assigned reading**? I think very likely so! As you can see, [this section of the assigned reading](https://guides.rubyonrails.org/active_record_basics.html#creating-active-record-models) does provide greater detail:

Screenshot from 2024-02-27 16-45-51

### Another problem
The other problem is I guess more psychological - - the sheer volume of the lesson content as it stands now. The preliminary reading is now more of a chore. And reading those docs all at once? I've done it, and I think it's a lot.

### Proposed solution
I would very much like to complete this task myself. I think the correct course of action is to split up the lesson, because that way it doesn't override the previous decision to keep the layout of the lessons consistent. That should make everyone happy -- the lesson layouts would be consistent, and the doc reading assignments would be correctly interleaved between the lesson text as was intended by the original author.

### Obstacles implementing the solution
Ok, I realize what I'm proposing is a big job and an extreme change. The new lessons would need to be formatted correctly -- they would need to have the proper sections, like Introduction and so on. So the content of each lesson would need to be re-arranged and edited pretty drastically. That's a lot, but I'm willing to give it a shot. Any content modification would of course be subject to everyone else's approval. It's conceivable that the job is too much for one person, in which case it would need to be split up. But I'd like to do it all myself if possible.

Aside from the writing, I don't actually have a full understanding of what else I'd need to do because I've never made a contribution as significant as adding new lessons. But, I could use [the guide](https://github.com/TheOdinProject/.github/blob/92e0734abfb36110d140c31f6cc9671935fc2862/CONTRIBUTING.md), [the wiki](https://github.com/TheOdinProject/theodinproject/wiki), fork and clone the repo, etc, and figure things out from there.

So, what do you guys think of my proposal? Is it alright if I do it?

### Path

Ruby / Rails

### Lesson Url

https://www.theodinproject.com/lessons/ruby-on-rails-active-record-basics

### (Optional) Discord Name

leo_jj

### (Optional) Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by Leo-U**
## Because

Completes the process of splitting Active Record Basics into multiple lessons.

## This PR

In `migrations.md`:
- Divides migrations content between headers.
- Adds additional content.
- Edits existing content.

In `active_record_basics.md`:
- Removes migrations content, removes associations content, de-lints.

In `basic_associations.md`:
- Inserts said associations content.

## Issue

Closes #27477

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Comment by Leo-U** [comment]
My mistake -- I forgot to wait for the validations lesson to get added the main repo before opening this PR. I should just leave this PR up and wait for that to happen first?

**Review by JoshDevHub — APPROVED** [review]
Looks great.

I'll have to wait until the Validations one is reviewed and merged before I merge this and put it on the main site.

Thank you for your contributions with this :rocket:

**Pull request merged**
